shared/modules/persistence_module.go

```diff
@@ -123,15 +125,16 @@ type PersistenceReadContext interface {
 	// Closes the read context
 	Close() error
 
+	// CONSOLIDATE: BlockHash / AppHash / StateHash
 	// Block Queries
-	GetPrevAppHash() (string, error) // app hash from the previous block
+	GetPrevAppHash() (string, error) // hash from the previous block relative to the context height
 	GetLatestBlockHeight() (uint64, error)
-	GetBlockHash(height int64) ([]byte, error)
+	GetBlockHashAtHeight(height int64) ([]byte, error)
 	GetBlocksPerSession(height int64) (int, error)
-	GetLatestProposerAddr() []byte
-	GetLatestBlockProtoBytes() []byte
-	GetLatestBlockHash() string
-	GetLatestBlockTxs() [][]byte
+	GetProposerAddr() []byte
+	GetBlockProtoBytes() []byte
+	GetBlockHash() string
```

Comment:
Any reason why `GetBlockHashAtHeight` returns []byte and `GetBlockHash` returns string?

Reply:
Just a result of all the string / byte inconsistencies we have throughout the codebase.

In #284, I consolidate it to: `GetBlockHash(height int64) ([]byte, error)`